lib/ansible/modules/system/aix_inittab.py

```diff
@@ -0,0 +1,239 @@
+                           'powerwait',
+                           'off',
+                           'hold',
+                           'ondemand',
+                           'initdefault',
+                           'sysinit'
+                           ], type='str'),
+            command = dict( required=True, type='str' ),
+            insertafter = dict( type='str' ),
+            state = dict( choices=[
+                          'present',
+                          'absent',
+                          ], required=True, type='str'),
+        )
+        )
+
+    result = {
+        'name':  module.params['name'],
+        'changed': False,
+        'status': {},
+        'msg': [],
+    }
+
+    # Find commandline strings
+    lsitab = module.get_bin_path('lsitab')
+    mkitab = module.get_bin_path('mkitab')
+    rmitab = module.get_bin_path('rmitab')
+
+    # check if the new entry exists
+    current_entry = check_current_entry(module)
+
+
+    # if action is install or change,
+    if module.params['state'] == 'present':
+
+        # create new entry string
+        new_entry = module.params['name']+":"+module.params['runlevel']+":"+module.params['action']+":"+module.params['command']
+
+        # If current entry exists or fields are different(if the entry does not exists, then the entry wil be created
+        if ( not current_entry['exist'] ) or (
+                module.params['runlevel'] != current_entry['runlevel'] or
+                module.params['action'] != current_entry['action'] or
+                module.params['command'] != current_entry['command'] ):
+
+            # If the entry does exist then change the entry
+            if current_entry['exist']:
+                (rc, out, err) = module.run_command(['chitab', new_entry])
+                if rc != 0:
+                    module.warn('could not change (%s)' % current_entry['name'])
+                result['status'] = "changed inittab entry"+" "+current_entry['name']
+                result['changed'] = True
+
+            # If the entry does not exist create the entry
+            elif not current_entry['exist']:
+
+                if module.params['insertafter']:
+                    (rc, out, err) = module.run_command(['mkitab', '-i',  module.params['insertafter'], new_entry ])
+                else:
+                    (rc, out, err) = module.run_command(['mkitab', new_entry ])
+
+                if rc != 0:
+                    module.warn('could not add (%s)' % current_entry['name'])
+                result['status'] = "add inittab entry"+" "+module.params['name']
+                result['changed'] = True
+
+
+    elif module.params['state'] == 'absent':
+        # If the action is remove and the entry exists then remove the entry
+        if current_entry['exist']:
+            (rc, out, err) = module.run_command(['rmitab',  module.params['name']])
+            if rc != 0:
+                module.warn('could not remove (%s)' % current_entry['name'])
```

Comment:
Looking at the logic, I am confused by why we are only sending out a warning when an entry cannot be added.
The correct action here is to fail.